crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -48,67 +109,160 @@ impl TryFrom<&MifinityRouterData<&types::PaymentsAuthorizeRouterData>> for Mifin
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
-                })
+            domain::PaymentMethodData::Wallet(wallet_data) => match wallet_data {
+                domain::WalletData::Mifinity(data) => {
+                    let money = Money {
+                        amount: item.amount.clone(),
+                        currency: item.router_data.request.currency.to_string(),
+                    };
+                    let phone_details = item.router_data.get_billing_phone()?;
+                    let client = MifinityClient {
+                        first_name: item.router_data.get_billing_first_name()?,
+                        last_name: item.router_data.get_billing_last_name()?,
+                        phone: phone_details.get_number()?,
+                        dialing_code: phone_details.get_country_code()?,
+                        nationality: item.router_data.get_billing_country()?,
+                        email_address: item.router_data.get_billing_email()?,
+                        dob: data.dob.clone(),
+                    };
+                    let address = MifinityAddress {
+                        address_line1: item.router_data.get_billing_line1()?,
+                        country_code: item.router_data.get_billing_country()?,
+                        city: item.router_data.get_billing_city()?,
+                    };
+                    let validation_key = format!(
+                        "payment_validation_key_{}_{}",
+                        item.router_data.merchant_id,
+                        item.router_data.connector_request_reference_id.clone()
+                    );
+                    let client_reference = item.router_data.customer_id.clone().ok_or(
+                        errors::ConnectorError::MissingRequiredField {
+                            field_name: "client_reference",
+                        },
+                    )?;
+                    let destination_account_number = data.destination_account_number;
+                    let trace_id = item.router_data.connector_request_reference_id.clone();
+                    let brand_id = Secret::new(get_brand_id_for_mifinity(
+                        &item.router_data.connector_meta_data,
+                    )?);
+                    Ok(Self {
+                        money,
+                        client,
+                        address,
+                        validation_key,
+                        client_reference,
+                        trace_id: Secret::new(trace_id.clone()),
+                        description: trace_id.clone(),
```

Comment:
Why are we not using payment description?

Reply:
Connector recommend to use the traceId in description for a better experience in the BackOffice application later.